### ISSUE TITLE
Shifting dependency from remote git to local editable does not work

### ISSUE THREAD

**Issue by dpompeu-xgeeks** [python-poetry/poetry#5743]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- OS: Windows 10 but using WSL with Ubuntu 20.04,4 LTS: 
- Poetry version: 1.1.3: 
- **[pyproject.toml](https://raw.githubusercontent.com/dpompeu-xgeeks/example-app/main/pyproject.toml) file**:

## Issue

I have some libraries and applications: the applications import the libraries, which are hosted on git. While debugging a library, I'd like to use a local version temporarily in editable mode, such that I can quickly get the changes and test them. 

However, this does not work as expected.

When shifting to _files_ and _develop_ and re-installing, poetry reads the new files and all the messages seem to indicate a successful change, but when running the application the changes are not there.

Strangely, changing from remote git to to _local_ mode **without the development flag**, works as expected, but not really fit for my workflow.

## Reproduction

```bash
$ poetry add git+ssh://git@github.com:dpompeu-xgeeks/example-lib.git#cad773cb633c3d8be5ddd16ef6767d5d0bc6e7cc
$ cat .venv/lib/python3.8/site-packages/mylib/mylib.py
$ poetry run python myapp.py
Hello World from git tag
```

```bash
$ echo 'print("Hello world from local dev")' > ../mylib/mylib/mylib.py
$ vim ../mylib/pyproject.toml # bump version (not sure if needed)
$ vim pyproject.toml # mylib = {path = "../mylib/", develop = true }
$ poetry lock; poetry install
Updating dependencies
Resolving dependencies... (0.1s)

Writing lock file
Installing dependencies from lock file

Package operations: 0 installs, 1 update, 0 removals

  • Updating mylib (0.1.1 -> 0.2.0 /home/dpompeu/dev/experiments/poetry-experiment/mylib)

Installing the current project: myapp (0.1.0)
$ poetry run python myapp.py 
Hello World from git tag # <----- no changes!
$ cat .venv/lib/python3.8/site-packages/mylib/mylib.py 
print("Hello World from git tag") # <---- no changes!
```

Removing the virtual env and re-installing:

```bash
$ rm -rf .venv/
$ poetry lock; poetry install
Creating virtualenv myapp in /home/dpompeu/dev/experiments/poetry-experiment/myapp/.venv
Updating dependencies
Resolving dependencies... (0.1s)
Installing dependencies from lock file

Package operations: 1 install, 0 updates, 0 removals

  • Installing mylib (0.2.0 /home/dpompeu/dev/experiments/poetry-experiment/mylib)

Installing the current project: myapp (0.1.0)
$ poetry run python myapp.py 
Hello world from local dev # <----- has changed!
```

Strangely, if we just shift from remote git to local (not editable) it works, but is not editable:

```bash
$ echo 'print("Hello world from local")' > ../mylib/mylib/mylib.py
$ vim ../mylib/pyproject.toml # bump version (not sure if needed)
$ vim pyproject.toml # mylib = {path = "../mylib/", develop = true }
$ poetry lock; poetry install
Updating dependencies
Resolving dependencies... (0.1s)

Writing lock file
Installing dependencies from lock file

Package operations: 0 installs, 1 update, 0 removals

  • Updating mylib (0.1.1 -> 0.2.0 /home/dpompeu/dev/experiments/poetry-experiment/mylib)

Installing the current project: myapp (0.1.0)
$ poetry run python myapp.py
Hello world from local # <----- has changed without additional steps!
```

**Comment by dpompeu-xgeeks** [comment]
There are more issues related to local development, such as #2725, but I haven't found one matching my exact use case.

**Comment by dpompeu-xgeeks** [comment]
Thanks for your support @abn, this worked as intended!

I'm still wondering if it's a result of the `--no-cache`, or from the bump from version `1.1` to `1.2`. I'll need to explore that further, and will update the issue.

If it's still useful, I'll leave two gists below: 

- [Fresh install with dependency from remote git](https://gist.github.com/dpompeu-xgeeks/5a78feac5db44c3cae54938f670a1302)
- [Update dependency by setting it to local and editable](https://gist.github.com/dpompeu-xgeeks/2d76c86725dfd34a0ada49c100a0f3a9)

And the output from running the application:

```bash
# right after installing
$ poetry@git run python myapp.py 
Hello World from git tag

# change to local file and modify it
$ echo 'print("Hello world from local dev")' > ../my-lib/mylib/mylib.py
$ poetry@git run python myapp.py
Hello world from local dev # <--- working as intended!

# modify again
$ echo 'print("Hello world from local dev 2")' > ../mylib/mylib/mylib.py 
$ poetry@git run python myapp.py 
Hello world from local dev 2 # <--- working as intended!

```

**Comment by dpompeu-xgeeks** [comment]
I see that `--no-cache` is is not available in `poetry 1.1`. 

So I omitted that parameter in `poetry 1.2` and things work as expected too! 

```bash
$ poetryx install
Creating virtualenv myapp in /home/duarte/dev/experiments/xpoetry/myapp/.venv
Installing dependencies from lock file

Package operations: 1 install, 0 updates, 0 removals

  • Installing mylib (0.1.1 cad773c)

Installing the current project: myapp (0.1.0)
$ poetryx run python myapp.py 
Hello World from git tag # <--- original print, as intended

$ vim pyproject.toml 
$ cat pyproject.toml | grep mylib
mylib = {path = "../mylib", develop=true} 
$ vim pyproject.toml 
$ poetryx install
Installing dependencies from lock file
Warning: poetry.lock is not consistent with pyproject.toml. You may be getting improper dependencies. Run `poetry lock [--no-update]` to fix it.

Package operations: 0 installs, 1 update, 0 removals

  • Updating mylib (0.1.1 cad773c -> 0.1.1 /home/duarte/dev/experiments/xpoetry/mylib)

Installing the current project: myapp (0.1.0)
$ echo 'print("Hello world from local dev")' > ../mylib/mylib/mylib.py     
$ poetryx run python myapp.py 
Hello world from local dev # <--- changed as intended!

$ echo 'print("Hello world from local dev 2")' > ../mylib/mylib/mylib.py 
$ poetryx run python myapp.py  # <--- changed as intended!
Hello world from local dev 2
```

**Comment by dpompeu-xgeeks** [comment]
I just installed `1.2.0` and it fixes this issue, so I'll close it.
